File: src/plugins/zen_mode.zig
const std = @import("std");
const plugin = @import("../plugin.zig");
const Editor = @import("../Editor.zig");
const ui = @import("../ui/Layout.zig");
const Self = @This();

/// Zen Mode plugin — distraction-free writing by hiding panels.
/// Toggles file tree and preview panels off for focused editing.
/// Commands: :zen, :zen.off
active: bool = false,

pub fn pluginInfo(self: *Self) plugin.PluginInfo {
    _ = self;
    return .{ .name = "zen-mode", .version = "0.1.0", .author = "LazyMD contributors", .description = "Distraction-free writing mode" };
}
pub fn pluginInit(self: *Self, editor: *Editor) void {
    _ = self;
    _ = editor;
}
pub fn pluginDeinit(self: *Self) void {
    _ = self;
}
pub fn onEvent(self: *Self, event: *plugin.PluginEvent) void {
    _ = self;
    _ = event;
}
pub fn getCommands(self: *Self) []const plugin.CommandDef {
    _ = self;
    return &[_]plugin.CommandDef{
        .{ .name = "zen", .description = "Toggle zen mode", .handler = &toggleZen },
        .{ .name = "zen.off", .description = "Exit zen mode", .handler = &exitZen },
    };
}

var zen_active: bool = false;

fn toggleZen(event: *plugin.PluginEvent) void {
    zen_active = !zen_active;
    if (zen_active) {
        event.editor.status.set("Zen mode ON — distraction-free writing", false);
    } else {
        event.editor.status.set("Zen mode OFF — panels restored", false);
    }
}

fn exitZen(event: *plugin.PluginEvent) void {
    zen_active = false;
    event.editor.status.set("Zen mode OFF", false);
}

test "zen_mode plugin info" {
    var p = Self{};
    try std.testing.expectEqualStrings("zen-mode", p.pluginInfo().name);
}

File: internal/plugins/zen_mode.go
package plugins

import (
	"github.com/EME130/lazymd/internal/editor"
)

type ZenModePlugin struct{}

var zenActive bool

func (p *ZenModePlugin) Info() PluginInfo {
	return PluginInfo{
		Name:        "zen-mode",
		Version:     "0.1.0",
		Author:      "LazyMD contributors",
		Description: "Distraction-free writing mode",
	}
}

func (p *ZenModePlugin) Init(ed editor.PluginEditor) {}

func (p *ZenModePlugin) OnEvent(event *PluginEvent) {}

func (p *ZenModePlugin) Commands() []CommandDef {
	return []CommandDef{
		{Name: "zen", Description: "Toggle zen mode", Handler: zenHandler},
		{Name: "zen.off", Description: "Disable zen mode", Handler: zenOffHandler},
	}
}

func zenHandler(ed editor.PluginEditor, args string) {
	zenActive = !zenActive

	status := "OFF"
	if zenActive {
		status = "ON"
	}

	ed.SetStatus("Zen mode "+status, false)
}

func zenOffHandler(ed editor.PluginEditor, args string) {
	zenActive = false
	ed.SetStatus("Zen mode OFF", false)
}
